chore(docs): docs structure refactor

signorecello opened the pull request:
Part 1 of the big docs refactor:

- Moving Learn into a Dropdown "Concepts" section
- Moving guides, tutorials, reference to their dropdown too
- Fixing all the broken links that came with it
- Adding redirects

signorecello commented:
* **#17509** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17509?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17185** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17185?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17161** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17161?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17161?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

critesjosh commented on docs/docusaurus.config.js:
```diff
@@ -185,10 +185,9 @@ const config = {
             position: "left",
```
The line above this points to `aztec/index`, which has been moved as part of this PR. When I click "Learn" in the navbar when I am on the `dev` branch, it takes me to v2.0.2. How should we handle this?
<img width="1018" height="212" alt="image" src="https://github.com/user-attachments/assets/9ac41702-7a97-497f-ba3f-b3397d3e92af" />

signorecello replied:
The only solution for `dev` would be to make it point to `docs.aztec.network/dev` which brings the correct page in `dev`. But because `docusaurus.config` isn't versioned, that link would always point to `/dev`. A user browsing the `2.0.2` version would be tricked into navigating the `/dev` version

critesjosh replied:
ok so do we need an ugly conditional check to route to the correct place? Then we could remove it after v2.0.2 goes away?

signorecello replied:
swizzling the navbar should work, on it

critesjosh commented on docs/docs/developers/docs/guides/local_env/advanced/faceid_wallet.md:
This section needs a `_category_.json`, the dropdown is showing "advanced", all lowercase.

<img width="581" height="675" alt="image" src="https://github.com/user-attachments/assets/56d24493-d925-439b-9e00-649cc7f0afa5" />

critesjosh reviewed (changes requested):
I see you renamed the `smart-contracts` folder to `aztec-nr`. This name is going to change again relatively soon (in the next couple of months), so I think we should keep it as `smart-contracts` for now to reduce the number of file path (and URL) changes.

critesjosh commented on docs/docs/developers/docs/concepts/smart_contracts/contract_creation.md:
This `smart_contracts` folder needs a `_category_.json` with a proper folder name
<img width="562" height="147" alt="image" src="https://github.com/user-attachments/assets/d254dfe4-7f48-49fa-97c8-9a9cbd340a3b" />

critesjosh commented on docs/docs/developers/docs/guides/smart_contracts/_category_.json:
sorry to be nit picky on this, but the `smart-contracts` folder has been renamed from `smart_contracts`. Can we keep it as `smart_contracts` so the urls dont break?

critesjosh reviewed (commented):
This is a better solution than the navbar swizzle. Just making sure that we want all of these expanded in v2.0.2 vs future versions.

<img width="576" height="1865" alt="image" src="https://github.com/user-attachments/assets/5b29639c-987e-46d3-9772-04791760ee8c" />

<img width="578" height="1040" alt="image" src="https://github.com/user-attachments/assets/7df37ee3-3f0b-4427-83da-c6eb075e292b" />

Also we should backport these changes to the v2 branch, right?

signorecello commented:
That was a halfway solution to avoid messing up with the whole structure on v2.0.2 as well... But I'll do it so they match

critesjosh commented:
just looking at the merge conflicts, maybe just forget about updating the nightly release since it will be updated shortly after we merge this anyways

signorecello committed:
chore(docs): docs structure refactor

Part 1 of the big docs refactor:

- Moving Learn into a Dropdown "Concepts" section
- Moving guides, tutorials, reference to their dropdown too
- Fixing all the broken links that came with it
- Adding redirects

Co-authored-by: José Pedro Sousa <outgoing@zpedro.dev>
